runtime/runc_jailer.go

```diff
@@ -466,9 +467,20 @@ func (j *runcJailer) setDefaultConfigValues(cfg *config.Config, socketPath strin
 // Close will cleanup the container that may be left behind if the jailing
 // process was killed via SIGKILL.
 func (j *runcJailer) Close() error {
-	return j.runcClient.Delete(j.ctx, j.vmID, &runc.DeleteOpts{
-		Force: true,
-	})
+	// Even the jailer's associated context is cancelled,
+	// we'd like to do the cleanups below just in case.
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	_, stateErr := j.runcClient.State(ctx, j.vmID)
+	if stateErr == nil {
+		// Delete the container, if it is still running.
+		err := j.runcClient.Delete(ctx, j.vmID, &runc.DeleteOpts{Force: true})
```

Comment:
I do have a little concern here about potentially leaking containers, since `State` can fail with any number of reasons. How do you feel about always trying to delete even if `stateErr` is `nil`

Reply:
I couldn't recall the reason of adding this check. Let me remove State() here and see how it works.